classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.inc.php

```diff
@@ -0,0 +1,134 @@
+                DB::table('review_round_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                error_log("Removing review files that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('review_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                // Create a Filesystem object with the appropriate adapter to access the actual files
+                $umask = Config::getVar('files', 'umask', 0022);
+                $adapter = new LocalFilesystemAdapter(
+                    Config::getVar('files', 'files_dir'),
+                    PortableVisibilityConverter::fromArray([
+                        'file' => [
+                            'public' => self::FILE_MODE_MASK & ~$umask,
+                            'private' => self::FILE_MODE_MASK & ~$umask,
+                        ],
+                        'dir' => [
+                            'public' => self::DIRECTORY_MODE_MASK & ~$umask,
+                            'private' => self::DIRECTORY_MODE_MASK & ~$umask,
+                        ]
+                    ]),
+                    LOCK_EX,
+                    LocalFilesystemAdapter::DISALLOW_LINKS
+                );
+
+                $filesystem = new Filesystem($adapter);
+
+                if ($filesystem->has($submissionFilePath)) {
+                    error_log("A submission file was found at ${submissionFilePath}. Trying to delete it...");
+                    try {
+                        $filesystem->delete($submissionFilePath);
+                        error_log("A submission file at ${submissionFilePath} was successfully deleted.");
+                    } catch (FilesystemException | UnableToDeleteFile $exception) {
+                        $exceptionMessage = $exception->getMessage();
+                        error_log("A submission file was found at ${submissionFilePath} but could not be deleted because of: ${exceptionMessage}.");
+                    }
+                } else {
+                    error_log("A submission file was expected but not found at ${submissionFilePath}.");
+                }
+                
+
+                DB::table('submission_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                DB::table('files')
+                    ->where('file_id', '=', $submissionFileFileId)
+                    ->delete();
+            }
+
+            error_log("Removing orphaned note entry ID ${noteId} with not existing query ${neQueryId}");
+            DB::table('notes')
+                ->where('note_id', '=', $noteId)
+                ->delete();
+        }
+    }
+
+    public function down(): void
+    {
+        throw new DowngradeNotSupportedException('Downgrade unsupported due to removed data');
```

Comment:
The data that has been removed was already orphaned data; throwing an exception here means that the 3.4.x to 3.3.x downgrade process will never complete successfullly. I would favour using an `error_log` in place of an exception, and letting the process continue. (We'll have to figure out some general policy for this.)

Reply:
Changed to `error_log`.

Just as a note here, I followed other examples like `https://github.com/pkp/pkp-lib/blob/main/classes/migration/upgrade/v3_4_0/I7190_UpdateFilters.php#L42` (and elsewhere) (which may be different cases of course)